src/state/multicall/hooks.ts

```diff
@@ -162,21 +163,31 @@ export function useSingleContractMultipleData(
 ): CallState[] {
   const fragment = useMemo(() => contract?.interface?.getFunction(methodName), [contract, methodName])
 
-  const blocksPerFetch = options?.blocksPerFetch
+  // encode callDatas
+  const callDatas = useMemo(
+    () =>
+      callInputs.every((callInput) => isValidMethodArgs(callInput)) && fragment
+        ? callInputs.map<string>((callInput) =>
+            // this type assertion is ok, because if fragment is defined, contract must be as well
```

Comment:
i don't love these because while true today they may not be true if someone edits the above fragment computation, so i would just add `&& contract` after `&& fragment`